### PR TITLE
Implement MemsetEnterScope

### LINKED ISSUE #430: MemsetEnterScope
Enters a new scope for the memory set operation with a specified value.

Reference: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/memset.cairo#L11

### PR DESCRIPTION
**Pull request by har777**
Fixes #430

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_others.go [hunk 1]
```diff
@@ -82,30 +82,48 @@ func createVMExitScopeHinter() (hinter.Hinter, error) {
...
+		name = "MemcpyEnterScope"
+	}
 	return &GenericZeroHinter{
-		Name: "MemcpyEnterScope",
+		Name: name,
 		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
-			//>  vm_enter_scope({'n': ids.len})
-			len, err := hinter.ResolveAsFelt(vm, len)
+			// MemsetEnterScope
+			//> vm_enter_scope({'n': n})
```
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_others.go:101]: A detail on this comment: The value assigned here is `ids.n`. For a moment I thought we'd have to change the code 😅 , until I checked with the [source code](https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/memset.cairo#L11)

Comment (har777) [reply pkg/hintrunner/zero/zerohint_others.go:101]: Ah bad comment 🥲

**Comment by cicr99** [comment]
Please @har777  review comments on #434, they apply on this PR as well

**Review by TAdev0 — APPROVED** [review]
lgtm!

**Review by cicr99 — APPROVED** [review]

**Pull request merged**